src/libCom/osi/os/WIN32/osdThread.c

```diff
@@ -526,6 +526,14 @@ static win32ThreadParam * epicsThreadParmCreate ( const char *pName )
         pParmWIN32->pName = (char *) ( pParmWIN32 + 1 );
         strcpy ( pParmWIN32->pName, pName );
         pParmWIN32->isSuspended = 0;
+#ifdef CREATE_WAITABLE_TIMER_HIGH_RESOLUTION
+        pParmWIN32->timer = CreateWaitableTimerEx(NULL, NULL, CREATE_WAITABLE_TIMER_HIGH_RESOLUTION, TIMER_ALL_ACCESS);
+        if (pParmWIN32->timer == NULL) {
+            pParmWIN32->timer = CreateWaitableTimer(NULL, 0, NULL);
+        }
+#else
+        pParmWIN32->timer = CreateWaitableTimer(NULL, 0, NULL);
+#endif
```

Comment:
Need some check if `CreateWaitableTimer()` can fail.  Also, since `pParmWIN32` is allocated with `calloc()`, this could be slightly shorten to:

```c
#ifdef CREATE_WAITABLE_TIMER_HIGH_RESOLUTION
        pParmWIN32->timer = CreateWaitableTimerEx(NULL, NULL, CREATE_WAITABLE_TIMER_HIGH_RESOLUTION, TIMER_ALL_ACCESS);
#endif
        if (pParmWIN32->timer == NULL) {
            pParmWIN32->timer = CreateWaitableTimer(NULL, 0, NULL);
        }
        if (pParmWIN32->timer == NULL) {
            cantProceed("CreateWaitableTimer()");
        }
```

Reply:
Thanks for the shortening. Should it return `NULL` on error (plus free memory) rather than calling `cantProceed` for consistency, or instead be calling `callocMustSucceed` rather than `calloc` earlier?

Reply:
ya ya ya.  I wasn't paying attention to the context...